src/id3/v2/items/key_value_frame.rs

```diff
@@ -0,0 +1,71 @@
+use crate::error::Result;
+use crate::id3::v2::frame::content::verify_encoding;
+use crate::id3::v2::Id3v2Version;
+use crate::util::text::{decode_text, encode_text, TextEncoding};
+
+use byteorder::ReadBytesExt;
+
+use std::io::Read;
+
+/// An `ID3v2` key-value frame
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct KeyValueFrame {
+	/// The encoding of the text
+	pub encoding: TextEncoding,
+	/// The key value pairs. Keys can be specified multiple times
+	pub key_value_pairs: Vec<(String, String)>,
+}
+
+impl KeyValueFrame {
+	/// Read an [`KeyValueFrame`] from a slice
+	///
+	/// NOTE: This expects the frame header to have already been skipped
+	///
+	/// # Errors
+	///
+	/// * Unable to decode the text
+	///
+	/// ID3v2.2:
+	///
+	/// * The encoding is not [`TextEncoding::Latin1`] or [`TextEncoding::UTF16`]
+	pub fn parse<R>(reader: &mut R, version: Id3v2Version) -> Result<Option<Self>>
+	where
+		R: Read,
+	{
+		let Ok(encoding_byte) = reader.read_u8() else {
+			return Ok(None);
+		};
+
+		let encoding = verify_encoding(encoding_byte, version)?;
+
+		let mut values = vec![];
```

Comment:
```suggestion
		let mut values = Vec::new();
```
This is just a nitpick, but I prefer `Vec::new()` over empty `vec!` calls.